sycl/source/device_selector.cpp

```diff
@@ -15,76 +15,133 @@
+      return true;
+    }
+  }
+  return false;
+}
+
 device device_selector::select_device() const {
   vector_class<device> devices = device::get_devices();
   int score = -1;
   const device *res = nullptr;
-  for (const auto &dev : devices)
-    if (score < operator()(dev)) {
+  for (const auto &dev : devices) {
+    int dev_score = operator()(dev);
+    if (detail::pi::trace(detail::pi::TraceLevel::PI_TRACE_ALL)) {
+      std::cout << "SYCL_PI_TRACE[1]: select_device(): -> score = " << score
+                << std::endl
+                << "SYCL_PI_TRACE[1]:   platform: "
+                << dev.get_info<info::device::platform>()
+                       .get_info<info::platform::version>()
+                << std::endl
+                << "SYCL_PI_TRACE[1]:   device: "
+                << dev.get_info<info::device::name>() << std::endl;
+    }
+
+    if (score < dev_score) {
       res = &dev;
-      score = operator()(dev);
+      score = dev_score;
     }
+  }
 
-  if (res != nullptr)
+  if (res != nullptr) {
+    if (detail::pi::trace(detail::pi::TraceLevel::PI_TRACE_BASIC)) {
+      std::cout << "SYCL_PI_TRACE[1]: select_device(): ->" << std::endl
+                << "SYCL_PI_TRACE[1]:   platform: "
+                << res->get_info<info::device::platform>()
+                       .get_info<info::platform::version>()
+                << std::endl
+                << "SYCL_PI_TRACE[1]:   device: "
+                << res->get_info<info::device::name>() << std::endl;
+    }
     return *res;
+  }
 
   throw cl::sycl::runtime_error("No device of requested type available.",
                                 PI_DEVICE_NOT_FOUND);
 }
 
 int default_selector::operator()(const device &dev) const {
+  int Score = -1;
 
-  // Take note of the SYCL_BE environment variable when doing default selection
-  const char *SYCL_BE = std::getenv("SYCL_BE");
-  if (SYCL_BE) {
-    std::string backend = (SYCL_BE ? SYCL_BE : "");
-    // Taking the version information from the platform gives us more useful
-    // information than the driver_version of the device.
-    const platform platform = dev.get_info<info::device::platform>();
-    const std::string platformVersion =
-        platform.get_info<info::platform::version>();;
-    // If using PI_CUDA, don't accept a non-CUDA device
-    if (platformVersion.find("CUDA") == std::string::npos &&
-        backend == "PI_CUDA") {
-      return -1;
-    }
-    // If using PI_OPENCL, don't accept a non-OpenCL device
-    if (platformVersion.find("OpenCL") == std::string::npos &&
-        backend == "PI_OPENCL") {
-      return -1;
-    }
-  }
-
-  // override always wins
-  if (dev.get_info<info::device::device_type>() == detail::get_forced_type())
-    return 1000;
+  // Give preference to device of SYCL BE.
+  if (isDeviceOfPreferredSyclBe(dev))
+    Score = 50;
 
   if (dev.is_gpu())
-    return 500;
+    Score += 500;
+
+  if (dev.is_accelerator())
```

Comment:
IIRC we have removed accelerator from default selector, why are you returning it?

Reply:
Restored this by mistake, thank you, fixed.